server/apps/publish/formatters/newsml_1_2_formatter.py

```diff
@@ -0,0 +1,163 @@
+                 'framing or similar means, without AAP\'s prior written permission.  AAP and its licensors ' \
+                 'are not liable for any loss, through negligence or otherwise, resulting from errors or ' \
+                 'omissions in or reliance on AAP content.  The globe symbol and "AAP" are registered trade marks. ' \
+                 'Further this AAP content is supplied to the direct recipient pursuant to an Information Supply ' \
+                 'Agreement with AAP (AAP Information Supply Agreement).  The direct recipient has a non-exclusive, ' \
+                 'non-transferable right to display this AAP content in accordance with and subject to the ' \
+                 'terms of the AAP Information Supply Agreement.'
+    LIMITATIONS = 'The direct recipient must comply with the limitations specified in the AAP Information ' \
+                  'Supply Agreement relating to the AAP content including, without limitation, not permitting ' \
+                  'redistribution and storage of the content (outside the terms of the Agreement) and not ' \
+                  'permitting deep hyperlinking to the content, framing of the content on a web site, ' \
+                  'posting the content to usenet newsgroups or facilitating such actions.'
+    now = utcnow()
+
+    def format(self, article, destination, selector_codes=None):
+        try:
+
+            pub_seq_num = superdesk.get_resource_service('output_channels').generate_sequence_number(destination)
+
+            newsml = etree.Element("NewsML")
+            SubElement(newsml, "Catalog", {'Href': 'http://www.aap.com.au/xml-res/aap-master-catalog.xml'})
+            news_envelope = SubElement(newsml, "NewsEnvelope")
+            news_item = SubElement(newsml, "NewsItem")
+
+            self.__format_news_envelope(article, news_envelope, pub_seq_num)
+            self.__format_identification(article, news_item)
+            self.__format_news_management(article, news_item)
+            self.__format_news_component(article, news_item)
+
+            return pub_seq_num, self.XML_ROOT + etree.tostring(newsml).decode('utf-8')
+        except Exception as ex:
+            raise FormatterError.newml12FormatterError(ex, destination)
+
+    def __format_news_envelope(self, article, news_envelope, pub_seq_num):
+        SubElement(news_envelope, 'TransmissionId').text = str(pub_seq_num)
+        SubElement(news_envelope, 'DateAndTime').text = self.now.strftime('%Y%m%dT%H%M%S+0000')
+        SubElement(news_envelope, 'Priority', {'FormalName': article.get('priority', '')})
+
+    def __format_identification(self, article, news_item):
+        revision = self.__process_revision(article)
+        identification = SubElement(news_item, "Identification")
+        news_identifier = SubElement(identification, "NewsIdentifier")
+        SubElement(news_identifier, 'ProviderId').text = 'aap.com.au'
```

Comment:
I separated providerId as well as the copyright text to aap_settings.py file which I think can later be used for AAP specific settings.